src/runtime/plugin.client.ts

```diff
@@ -0,0 +1,144 @@
+import { globalName, storageKey } from '#color-mode-options'
+import { defineNuxtPlugin, isVue2, isVue3 } from '#app'
+import { addRouteMiddleware, useRoute, useRouter, useState } from '#imports'
+import { reactive, watch } from 'vue'
+
+import type { ColorModeInstance } from './types'
+
+const helper = window[globalName] as unknown as {
+  preference: string
+  value: string
+  getColorScheme: () => string
+  addClass: (className: string) => void
+  removeClass: (className: string) => void
+}
+
+const getForcedColorMode = route => route.matched[0] && route.matched[0].components.default.options.colorMode
+
+export default defineNuxtPlugin((nuxtApp) => {
+  const colorMode = useState<ColorModeInstance>('color-mode', () => reactive({
```

Comment:
Note: Maybe we could improve `useState` to ensure input objects remain reactive?

Reply:
It seems that the object is reactive by default @danielroe 

See https://stackblitz.com/edit/nuxt-starter-cll44i?file=app.vue

Reply:
The issue is that we are 'reaching in' to the object within `useState` in order to expose a reactive object. (`<template>` unwraps it automatically for us, which conceals the issue.)